src/tqec/circuit/detector/detector_util.py

```diff
@@ -0,0 +1,406 @@
+from __future__ import annotations
+
+import dataclasses
+from typing import Iterable
+
+import numpy as np
+import stim
+
+from tqec.exceptions import TQECException
+
+ANNOTATIONS = {
+    "QUBIT_COORDS",
+    "DETECTOR",
+    "OBSERVABLE_INCLUDE",
+    "SHIFT_COORDS",
+    "TICK",
+}
+
+
+class PauliString:
+    def __init__(self, qubits: dict[int, str]):
+        """A mapping from qubits to Pauli operators that represent a Pauli string.
+
+        Args:
+            qubits: A dictionary mapping qubit indices to Pauli operators. The
+                Pauli operators should be one of "I", "X", "Y", or "Z".
+        """
+        for qubit, pauli in qubits.items():
+            if pauli not in "IXYZ":
+                raise TQECException(
+                    f"Invalid Pauli operator {pauli} for qubit {qubit}, expected I, X, Y, or Z."
+                )
+        self.qubits = {q: qubits[q] for q in sorted(qubits.keys())}
+        self._hash = hash(tuple(self.qubits.items()))
```

Comment:
Maybe 
```suggestion
    def __init__(self, qubits2pauli: dict[int, str]):
        """A mapping from qubits to Pauli operators that represent a Pauli string.

        Args:
            qubits2pauli: A dictionary mapping qubit indices to Pauli operators. The
                Pauli operators should be one of "I", "X", "Y", or "Z".
        """
        for qubit, pauli in qubits2pauli.items():
            if pauli not in "IXYZ":
                raise TQECException(
                    f"Invalid Pauli operator {pauli} for qubit {qubit}, expected I, X, Y, or Z."
                )
        self.qubits = {q: qubits2pauli[q] for q in sorted(qubits2pauli.keys())}
        self._hash = hash(tuple(self.qubits.items()))
```
or another name that would be more expressive about the fact that the datastructure is not a simple list of qubits but rather a mapping from qubit indices to paulis?